packages/keyv/src/index.ts

```diff
@@ -0,0 +1,160 @@
+import EventEmitter from 'events';
+import JSONB from 'json-buffer';
+
+const adapters = {
+	redis: '@keyv/redis',
+	mongodb: '@keyv/mongo',
+	mongo: '@keyv/mongo',
+	sqlite: '@keyv/sqlite',
+	postgresql: '@keyv/postgres',
+	postgres: '@keyv/postgres',
+	mysql: '@keyv/mysql'
+};
+
+const loadStore = <
+	TVal,
+	TAdapterOpts extends {
+		adapter?: keyof typeof adapters;
+		uri?: string;
+	}
+>(opts: TAdapterOpts): KeyvStore<TVal> => {
+	/* eslint-disable @typescript-eslint/no-var-requires */
+	const validAdapters = Object.keys(adapters);
+
+	if (opts.adapter) {
+		if (validAdapters.includes(opts.adapter)) {
+			throw new Error(`[keyv]: Invalid adapter "${opts.adapter}"`);
+		} else {
+			try {
+				const adapter = require(adapters[opts.adapter]);
+				return new (adapter.__esModule ? adapter.default : adapter)(opts);
+			} catch (e) {
+				throw new Error(`[key]: Failed to load adapter "${opts.adapter}", did you forget to install it?\n\nnpm install @keyv/${opts.adapter}\n\n${e}`);
+			}
+		}
+	}
+
+	if (opts.uri) {
+		const matches = /^[^:]*/.exec(opts.uri);
+		if (matches === null || !Object.keys(adapters).includes(matches[0])) {
+			throw new Error(`[keyv]: Failed to load adapter with uri "${opts.uri}"`);
+		}
+
+		const adapterName = matches[0] as keyof typeof adapters;
+		const adapter = require(adapters[adapterName]);
+		return new (adapter.__esModule ? adapter.default : adapter)(opts);
+	}
+
+	return new Map() as (Map<string, string> & { namespace: string });
+	/* eslint-enable @typescript-eslint/no-var-requires */
+};
+
+type MaybePromise<T> = T | Promise<T>;
+
+export interface KeyvOptions<TVal, TSerialized = string> {
+	namespace: string;
+	store: KeyvStore<TVal, TSerialized>;
+	adapter?: keyof typeof adapters;
+	ttl?: number;
+	serialize(x: KeyvStoreObject<TVal>): TSerialized;
+	deserialize(x: TSerialized): KeyvStoreObject<TVal>;
+}
+
+export interface KeyvStore<TVal, TSerialized = string> {
+	namespace: string;
+	ttlSupport?: boolean;
+
+	on?(event: 'error', cb: (err: Error) => void | never): void;
+
+	get(key: string): MaybePromise<void | TSerialized | KeyvStoreObject<TVal>>;
+	set(key: string, val: TSerialized, ttl?: number): MaybePromise<unknown>;
+	delete(key: string): MaybePromise<boolean>;
+	clear(): MaybePromise<void>;
+}
+
+export interface KeyvStoreObject<TVal> {
+	expires: number | null;
+	value: TVal;
+}
+
+export type KeyvStoreEvent = 'error';
+
+export default class Keyv<TVal> extends EventEmitter {
+	protected readonly opts: KeyvOptions<TVal>;
+
+	constructor(uriOrOpts: string | Partial<KeyvOptions<TVal>>, _opts: Partial<KeyvOptions<TVal>> = {}) {
+		super();
+
+		const adapterOpts = ({
+			namespace: 'keyv',
+			serialize: JSONB.stringify,
+			deserialize: JSONB.parse,
+			...((typeof uriOrOpts === 'string') ? { uri: uriOrOpts } : uriOrOpts),
+			..._opts
```

Comment:
The linter changed these to object rest spreads. Personally I prefer this syntax, but if you aren't familiar, the following are equivalent

```javascript
const foo = Object.assign({}, obj1, obj2)

const foo = {
  ...obj1,
  ...obj2
}
```